### ISSUE TITLE
Second thoughts on BaseProvider.get_bare_instance()

### ISSUE THREAD

**Issue by andy-maier** [pywbem/pywbem#2298]
I've got second thoughts on the recent change of `BaseProvider.find_instance()` to `get_bare_instance()`.

For one, the name is just ugly.

More importantly though, the method gets an `instance_store` parameter, even though it is clear that it always would find the instance in the instance store for the namespace of the instance to be found, in the CIM repository that is known by the BaseProvider object. The namespace could be required to be set on the instance to be found (and is in fact set on all uses in the code).

**Comment by andy-maier** [comment]
Addressed by PR #2299.

**Comment by KSchopmeyer** [comment]
COMMENTS/KS

1. first history.  Did the find_instance as the original instance bare get early in mocker when the instance repo was a list and we searched the list to find the instance with the correct path.  At that point there were issues with duplication, etc. so named it find. What was logical then, just never got changed.  Thus, I agree that the word 'get' should be there

2. The word 'bare'   Dictionary has multiple meanings for bard but one is "basic or simple", bare metal, etc.  It is commmon, was actually used in original pywbem server implementation for just this and I find it other code.

3. This method also was based on the store because when used to get multiple instances, it avoids the extra load of namespace search for each item.

In short, I actually like the name and usage.  Whether we use it often enough in the code to justify all this would need more research.

**Comment by andy-maier** [comment]
On 1. and 2: I think I can accept the name `get_bare_instance()`.
On 3: Where exactly is there a namespace "search" now? The only thing in that direction is a lookup of the namespace in the dictionary of stores, and that happens before and after this change, just in a different place.

**Comment by KSchopmeyer** [comment]
Sorry did not mean search but lookup; while it was originally a search of the instances list that disappeared long ago. While only a few lines long to validate  the namespace, I dId not want to force this on the possible case of multiple get instances when the get_instance_store does that once os getting the bare instance is really just a dictionary lookup (namespace and path).

**Comment by andy-maier** [comment]
The method is part of the public interface for provider implementations, and it is not logical having to pass in the instance store when that is already determined by the other parameters.

The only loop I found where in the current code the instance store is determined once and then used multiple times in the loop, is the EnumerateInstances operation in the main provider. If you are concerned about the repeated dict access within the loop that could be outside, then let's make this method private for now. The main provider implementations use other private methods already.

**Comment by KSchopmeyer** [comment]
The main loop is actually in _get_associated_instancenames.

Here are the various usage:
_providerdispatcher.py  188:        orig_instance = self.get_bare_instance(ModifiedInstance.path,

replace with object exists, etc. or just change to use namespace since it is only once per operation.

_baseprovider.py
Only he definition, no usage.
461:    def get_bare_instance(instance_name, instance_store, copy=None):

_instancewriteprovider.py
497:        orig_instance = self.get_bare_instance(modified_instance.path,
Gotten and used. Since this is once per operation and write operations only there is no issue of performance if the interface is changed.

_mainprovider.py
404:        rtn_inst = self.get_bare_instance(instance_name, instance_store,
   Gets instance from repository to work on.  Could be replaced with direct call to repository or namespace version of same code
   since this is called with namespace parameter in any case.

1855:            inst = self.get_bare_instance(ref_path, instance_store)
  Loop looking for instances.  Replace with lookup directly into repository

_wbemconnection_mock.py
886:                    if self._mainprovider.get_bare_instance(inst.path,
  Existence test only - Replace with object_exists.

We completely disagree on this one since using the get_instance_store seems logical to me but I just want to get this beta out.

Go ahead and make minimal changes and I will  approve

**Comment by KSchopmeyer** [comment]
I just approved the pr that makes this private.

**Comment by andy-maier** [comment]
Ok, Thx.
